### PR TITLE
Setup instructions and development tools

### PR DESCRIPTION
**Pull request by felixwiemuth**

### CODE REVIEW

File: compiler/build.sh
Comment (aslanix) [inline compiler/build.sh:1]: I'm not sure I understand the purpose of this build script. Why is it needed?

File: compiler/.gitignore
Comment (aslanix) [inline compiler/.gitignore:1]: Parser.hs does not live in the src folder. Why add it here?

Comment (felixwiemuth) [reply compiler/.gitignore:1]: This is where happy generates `Parser.hs` (in the same folder as `Parser.y`).

File: compiler/build.sh
Comment (felixwiemuth) [reply compiler/build.sh:1]: It is useful when you want to compare different versions of the executables when making changes. It copies the build to a separate folder named after the git hash of the corresponding code state.

File: .vscode/tasks.json
[the remark this answers is not in the document]
Comment (felixwiemuth) [reply .vscode/tasks.json:1]: It is useful to have compilation as a task in VSCode which can be bound to a keybinding (here it's configured as the default build task).

File: compiler/.gitignore
Comment (felixwiemuth) [reply compiler/.gitignore:1]: Right, stack doesn't seem to generate the file. I reverted the commit.

File: compiler/build.sh
Comment (felixwiemuth) [reply compiler/build.sh:1]: Done.

**Comment by felixwiemuth** [comment]
(The reason some of the comments are not connected to code anymore is probably that they were written before but published after I force-pushed while there were no comments yet.)

**Review by aslanix — APPROVED** [review]

**Pull request merged**